### PR TITLE
Switch CI to GitHub Actions

### PR DESCRIPTION
**Pull request by garethbowen**
#20

### CODE REVIEW

**Comment by garethbowen** [comment]
@newtewt I think this solves the problem. Would you mind reviewing?

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+      matrix:
+        node-version: [8.x, 10.x, 12.x]
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Use Node.js ${{ matrix.node-version }}
+      uses: actions/setup-node@v1
+      with:
+        node-version: ${{ matrix.node-version }}
+    - name: Set up CouchDB
```
Comment (newtewt) [inline .github/workflows/ci.yml:45]: Could this be moved to the services containers instead of using a third party github action?

**Comment by newtewt** [comment]
Just one question about two approaches to the same thing on the containers. Otherwise looks good.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (garethbowen) [reply .github/workflows/ci.yml:45]: The 3rd party action does a few more steps that the docker image doesn't do by default, particularly creating the system dbs at the bottom of [the entrypoint script](https://github.com/cobot/couchdb-action/blob/master/entrypoint.sh). We could inline those but I'm happy using the third party action to save us maintaining those steps. What do you think?

Comment (newtewt) [reply .github/workflows/ci.yml:45]: Nope I think that's fine. One last request would be to use the commit hash on third party actions. It is recommended by the docs. Also using straight master seems  bit concerning.  I think this warrants a change to our docs too.  I'll put  PR up.

https://docs.github.com/en/actions/learn-github-actions/security-hardening-for-github-actions#using-third-party-actions

**Comment by garethbowen** [comment]
Nice one! Thanks for the suggestion.

**Pull request merged**